### DISCUSSION TITLE
LDAP authentication and Indexing

### DISCUSSION THREAD

**Discussion by sunilkalra123** [oracle/opengrok#4127] [Q&A] [answered]
Ive just installed tomcat 10 and the latest opengrok. 

I have configured LDAP authentication via tomcat using form based login. This seems to work fine for users. However, when indexing, it seems to have an issue because the api is also protected by the form based ldap authentication.

Is there a way to do the indexing without needing to use the API or webapp?

the error i get when running the index (as per opengrok documnetation) is 
`java.io.IOException: InboundJaxrsResponse{context=ClientResponse{method=PUT, uri=http://localhost:8080/source/api/v1/configuration?reindex=true, status=403, reason=Forbidden}}`

any ideas how i can get this to work? im guessing i will have to allow unauthenticated access to /api then set up token based authorization within opengrok for the indexer and opengrok ldap authorization for users.

**Comment by vladak** [comment, marked as answer]
> Is there a way to do the indexing without needing to use the API or webapp?

You need to poke a hole in the paths protected by the authentication to allow `/source/api/v1/`. Assuming the indexer is running on localhost and the `-U` is also using localhost in the URI.

**Reply by vladak** [reply]
See the configuration on https://github.com/oracle/opengrok/wiki/Authorization-based-on-HTTP-Basic-Authentication#application-deployment-descriptor for example w.r.t. said hole.

**Reply by vladak** [reply]
related: https://github.com/oracle/opengrok/discussions/4129
